File: DMOJ/ccc13j2.cpp
#include <iostream> 
#include <string> 

using namespace std; 

int main(){
    string word; 
    cin >> word; 
    int counter = 0;
    for (int i = 0; i < word.length(); i++){
        if (word[i] == 'I' || word[i] == 'O' || word[i] == 'S' || word[i] == 'H' || word[i] == 'Z' || word[i] == 'X' || word[i] == 'N'){
            counter ++; 
        } 
    }
    if (counter == word.length()){
        cout << "YES"; 
    }else { 
        cout << "NO";
    }
    return 0;
}

File: DMOJ/dmoj.ca-downloaded-submissions/ccc13j2.py
word = input()

allowed_letters = ['I', 'O', 'S', 'H', 'Z', 'X', 'N']

list_word = list(word)

counter = 0

for i in range(0, len(list_word)):
    if list_word[i] in allowed_letters:
        counter += 1
    
if counter == len(list_word):
    print("YES")
else:
    print("NO")